osu.Game/Database/ArchiveModelManager.cs

```diff
@@ -331,7 +331,7 @@ private string computeHash(ArchiveReader reader)
                             if (CanUndelete(existing, item))
                             {
                                 Undelete(existing);
-                                Logger.Log($"Found existing {typeof(TModel)} for {item} (ID {existing.ID}). Skipping import.", LoggingTarget.Database);
+                                Logger.Log($"Found existing {nameof(TModel)} for {item} (ID {existing.ID}). Skipping import.", LoggingTarget.Database);
```

Comment:
`nameof(TModel)` always resolves to "TModel". Is that intentional?

Reply:
It is not intentional. Also this PR is still a draft; there are multiple more passes I will be doing before it is ready for review.